source/common/http/conn_manager_impl.cc

```diff
@@ -404,6 +423,15 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
   // Set the trusted address for the connection by taking the last address in XFF.
   downstream_address_ = Utility::getLastAddressFromXFF(*request_headers_);
   decodeHeaders(nullptr, *request_headers_, end_stream);
+
+  Tracing::Decision tracing_decision =
```

Comment:
I'll double check that, but i had an impression that before decodeHeaders(..) we would not know whether request is HC or not for example